### PR TITLE
CB Fixes and Improvements

### LINKED ISSUE #8372: Issues in the Component Browser in the new IDE
- The `qnIsTopElement` function is filtering out all the methods that should be in the default list.
The code was changed from `(name.match(/\./g)?.length ?? 0) <= 2` to `!/[.].*?[.].*?[.]/.test(name)`.
These are not equivalent. The later only works if 3 parts in the name.
`name.split('.'),length <= 2` is probably clearer.

- We should remove modules, types and constructors from the top level list unless we have added a GROUP to them (though not even sure that works at moment).
```ts
  private mainViewFilter(entry: SuggestionEntry): MatchResult | null {
    const hasGroup = entry.groupIndex != null
    const isModule = entry.kind === SuggestionKind.Module
    const isType = entry.kind === SuggestionKind.Type
    const isConstructor = entry.kind === SuggestionKind.Constructor
    const isTopElement = qnIsTopElement(entry.definedIn)
    if (!hasGroup && (isModule || isType || isConstructor || !isTopElement)) return null
    else return { score: 0 }
  }
```

- The initial filtering should just be initials starts with not a letter by letter match.
For example, within a Table, `sc` should match `select_columns` but not `select_blank_columns`.

- Ordering is not being applied. The sort algorithm is missing in `matched`.
`const matched = Array.from(matchSuggestions()).sort(compareSuggestions)`

- We should only show the local variables if `showLocal` is `true`.

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

* The "main view" (when no self type/module nor pattern is specified) contains all methods defined in top modules of every library
* Entries defined inside any "Internal" module from stdlib will be hidden by default; they will be visible only when self type is specified or the module path leading to such module internals.
* Fixed an issue with missing groups sometimes - we must wait with asking for groups for first executionComplete message.
* Modules and local variables have different default icon.

image

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/ComponentBrowser/__tests__/filtering.test.ts [hunk 1]
```diff
@@ -20,16 +20,23 @@ test.each([
...
 test.each([
-  makeModuleMethod('Standard.Base.Data.convert'), // not in group
+  makeModuleMethod('Standard.Base.Data.Vector.some_method'), // not in top group
   { ...makeMethod('Standard.Base.Data.Vector.Vector.get'), groupIndex: 1 }, // not static method
   makeModule('Standard.Base.Data.Vector'), // Not top module
+  makeStaticMethod('Standard.Base.Internal.Foo.bar'), // Internal method
+  makeModule('Standard.Base.Internal'), // Internal module
+  makeModule('Standard.Internal.Foo'), // Internal project
 ])('$name entry is not in the CB main view', (entry) => {
+  console.log(entry.definedIn)
```
Comment (MichaelMauderer) [inline app/gui2/src/components/ComponentBrowser/__tests__/filtering.test.ts]: Missed debug logging.

**Review by MichaelMauderer — APPROVED** [review]

File: app/gui2/src/stores/suggestionDatabase/index.ts [hunk 2]
```diff
@@ -119,6 +120,7 @@ class Synchronizer {
       })
     })
     this.queue.pushTask(async ({ currentVersion }) => {
+      await nextEvent(lsRpc, 'executionContext/executionComplete')
```
Comment (Frizi) [inline app/gui2/src/stores/suggestionDatabase/index.ts]: Is this guaranteed to progress? What if the `executionComplete` event happened before this await?

**Review by Frizi — APPROVED** [review]

**Comment by somebody1234** [comment]
@farmaazon unrelated to PR, but should we be making it a habit to add [x-new-feature] to PRs when appropriate?

**Comment by farmaazon** [comment]
> @farmaazon unrelated to PR, but should we be making it a habit to add [x-new-feature] to PRs when appropriate?

Nice to have, but not obligatory (yet).

File: app/gui2/src/stores/suggestionDatabase/index.ts (hunk 2 above)
Comment (farmaazon) [reply app/gui2/src/stores/suggestionDatabase/index.ts]: > Is this guaranteed to progress? What if the `executionComplete` event happened before this await?

Good catch. We have to track this event even before creating first execution context.

**Pull request merged**